felix/dataplane/linux/bpf_ep_mgr.go

```diff
@@ -331,8 +330,14 @@ func newBPFEndpointManager(
 			254,
 			opReporter,
 		)
-		m.services = make(map[serviceKey][]string)
-		m.dirtyServices = make(map[serviceKey][]string)
+		// Since we do not know what the state of services is at the start, mark
+		// the routes to be all cleaned up. As the service update will arrive,
+		// we will update the state.
+		if err := m.invalidateServiceRoutes(); err != nil {
```

Comment:
I don't think you should even need this start-of-day clean up.  The RouteTable logic is built around the idea of "Calico owned routes".  It uses a matcher function to tell if a route belongs to us and if so, it automatically cleans it up if the manager layer doesn't tell it that the route should exist before the first Apply.

It's like how, to clear a caching map, all you need to do is create the caching map and then hit Apply(); it'll load the dataplane state and spot that the desired state is an empty map and clea up accordingly.

The only prereq for that is to have a good matcher function and I think you're in luck here since all your route point at a Calico-owned interface so if it points at that interface, it's ours.

Reply:
ok, wasnt sure about that behaviour, if it is so, great :+1: